runtime-modules/content/src/lib.rs

```diff
@@ -1261,6 +944,548 @@ decl_module! {
         ) {
             Self::not_implemented()?;
         }
+
+        /// Start video nft auction
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn start_nft_auction(
+            origin,
+            auctioneer: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+            auction_params: AuctionParams<T::VideoId, T::BlockNumber, BalanceOf<T>, T::MemberId>,
+        ) {
+
+            let video_id = auction_params.video_id;
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure nft is already issued
+            video.ensure_nft_is_issued::<T>()?;
+
+            // Ensure there nft transactional status is set to idle.
+            video.ensure_nft_transactional_status_is_idle::<T>()?;
+
+            // Authorize nft owner
+            Self::authorize_nft_owner(
+                origin,
+                &auctioneer,
+                &video
+            )?;
+
+            // Validate round_duration & starting_price
+            Self::validate_auction_params(&auction_params)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Create new auction
+            let auction = AuctionRecord::new(auction_params.clone());
+            let video = video.set_auction_transactional_status(auction);
+
+            // Update the video
+            VideoById::<T>::insert(video_id, video);
+
+            // Trigger event
+            Self::deposit_event(RawEvent::AuctionStarted(auctioneer, auction_params));
+        }
+
+        /// Cancel video auction
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn cancel_video_auction(
+            origin,
+            auctioneer: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+            video_id: T::VideoId,
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Authorize nft owner
+            Self::authorize_nft_owner(
+                origin,
+                &auctioneer,
+                &video
+            )?;
+
+            // Ensure auction for given video id exists
+            let auction = video.ensure_nft_auction_state::<T>()?;
+
+            // Ensure nft auction not expired
+            Self::ensure_nft_auction_not_expired(&auction)?;
```

Comment:
> The requirement for cancelling an auction, English or open, is that there are no active bids. I think there are some non-trivial problems with the other checks here, but its not worth going into.

Can't `Open` auction be canceled if there is a bid present already?

Redundant check removed 37bb1618b3

Reply:
> Can't Open auction be canceled if there is a bid present already?

I guess you are right that this actually makes sense, because the owner has to accept the bid anyway, so may as well have the right to just cancel the whole thing. But for English you cannot, as a bid is binding attempt to buy at given price unless someone outbids.